src/murmur/Server.cpp

```diff
@@ -844,21 +844,22 @@ void Server::run() {
 
 				MessageHandler::UDPMessageType msgType = static_cast<MessageHandler::UDPMessageType>((buffer[0] >> 5) & 0x7);
 
-				switch (msgType) {
-					case MessageHandler::UDPVoiceSpeex:
-					case MessageHandler::UDPVoiceCELTAlpha:
-					case MessageHandler::UDPVoiceCELTBeta:
-						if (bOpus)
-							break;
-					case MessageHandler::UDPVoiceOpus: {
-							u->aiUdpFlag = 1;
-							processMsg(u, buffer, len);
-							break;
-						}
-					case MessageHandler::UDPPing: {
-							QByteArray qba;
-							sendMessage(u, buffer, len, qba, true);
-						}
+				if (msgType == MessageHandler::UDPVoiceSpeex ||
+					msgType == MessageHandler::UDPVoiceCELTAlpha ||
+					msgType == MessageHandler::UDPVoiceCELTBeta ||
+					msgType == MessageHandler::UDPVoiceOpus) {
+
+					// Determine if the packet should be processed
+					// by Murmur. Clients can still send non-Opus
+					// voice packets when in Opus mode. However,
+					// Murmur should discard them in that case.
+					if (bOpus && msgType != MessageHandler::UDPVoiceOpus) {
```

Comment:
This logic doesn't make sense? "If we're in Opus mode, only process packets that are not Opus".

Reply:
The reason I split it up in the first place is to make the code easier to read.

I still prefer

    bool ok = true;
    if (bOpus && msgType != MessageHandler::UDPVoiceOpus) {
   		ok = false;
    }

This (at least to me), reads:
1. Allow all packets through by default.
2. Unless we're in Opus mode. In Opus mode, only Opus packets are allowed.

So by that logic, maybe we should refactor the comment instead?

    // Allow all voice packets through by default.
    bool ok = true;
    // ...Unless we're in Opus mode. In Opus mode, only Opus packets are allowed.
    if (bOpus && msgType != MessageHandler::UDPVoiceOpus) {
   		ok = false;
    }

Reply:
Oops... my fault. I read the inverted logic.